bin/ci.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+set -e
+
+##
+# Auxiliar functions
+
+PORTUS_DB_ADAPTER=${PORTUS_DB_ADAPTER:-mysql2}
+
+# Interacts with a daemon by taking two arguments:
+#   1. The action (e.g. "start").
+#   2. The service (e.g. "mysql").
+# We do this to abstract the fact that Travis CI does not use systemd and we do.
+function __daemon() {
+    if [[ -z "$CI" ]]; then
+        sudo systemctl $1 $2
+    else
+        sudo service $2 $1
+    fi
+}
+
+# Performs systemctl calls to the current database adapter when used outside of
+# a container.
+function __database() {
+    if [[ -f /.dockerenv ]]; then
+        return
+    fi
+
+    if [[ "$PORTUS_DB_ADAPTER" == "mysql2" ]]; then
+        __daemon $1 mysql
+    else
+        __daemon $1 postgresql
+    fi
+}
+
+# Setup an insecure registry for the local docker.
+function __docker_insecure() {
+    if [[ ! -z "$CI" ]]; then
+        sudo tee /etc/docker/daemon.json > /dev/null <<EOF
+{
+  "insecure-registries" : ["172.17.0.1:5000"]
+}
+EOF
+    fi
+    __daemon restart docker
+
+    # Show version info
+    docker --version
+    docker-compose --version
+}
+
+##
+# The actual run
+
+# Test commit messages
+bundle exec rake test:git
+
+# Style and security checks
+bundle exec rubocop -V
+bundle exec rubocop -F
+
+# Compile assets
+bundle exec rake portus:assets:compile
+
+# Ruby tests
+__database restart
+bundle exec rspec spec
+__database stop
```

Comment:
Any specific reason to restart and then stop the database?

Reply:
This is because unit tests require the DB to be up, and integration tests raise their own database, so we should stop the main DB so there are no clashes.